src/odoc/uri.ml

```diff
@@ -0,0 +1,54 @@
+open Result
+
+let resolve url_to_string directories reference prefix =
+  let resolver =
+    Resolver.create ~important_digests:false ~directories ~open_modules:[]
+  in
+  let reference =
+    let open Odoc_model in
+    let warnings_options = { Error.warn_error = true; print_warnings = true } in
+    Semantics.parse_reference reference
+    |> Error.handle_errors_and_warnings ~warnings_options
+  in
+  match reference with
+  | Error e -> Error e
+  | Ok reference -> (
+      let environment = Resolver.build_env_for_reference resolver in
+      let resolved_reference =
+        Odoc_xref2.Ref_tools.resolve_reference environment reference
+        |> Odoc_model.Error.raise_warnings
+      in
+      match resolved_reference with
+      | Error e ->
+          let error =
+            Format.asprintf "%a"
+              Odoc_xref2.Errors.Tools_error.pp_reference_lookup_error e
+          in
+          Error (`Msg error)
+      | Ok resolved_reference -> (
+          let identifier =
+            Odoc_model.Paths.Reference.Resolved.identifier resolved_reference
+          in
+          let url =
+            Odoc_document.Url.from_identifier ~stop_before:false identifier
+          in
+          match url with
+          | Error e -> Error (`Msg (Odoc_document.Url.Error.to_string e))
+          | Ok url ->
+              let href = url_to_string prefix url in
+              print_endline href;
+              Ok ()))
+
+let reference_to_uri_html =
+  let url_to_string prefix url =
+    let prefix = match prefix with None -> "" | Some prefix -> prefix in
+    Odoc_html.Link.(href ~resolve:(Base prefix) url)
+  in
+  resolve url_to_string
+
+let reference_to_uri_latex =
+  let url_to_string prefix url =
+    let prefix = match prefix with None -> "" | Some prefix -> prefix in
+    prefix ^ Odoc_latex.Generator.Link.label url
```

Comment:
I don't think `prefix` makes sense for latex. The `label` here is not a path but the equivalent of an anchor.
It's expected that the latex files are processed again, as a whole, and that the only things that survive this step are these labels and not the paths.

Reply:
Do you want us to make it optional or get rid of it entirely? The former makes sense to me?

Reply:
I think the prefix option shouldn't be allowed at all in "latex mode". I'm not sure whether that's possible with cmdliner though.
Perhaps it can be split into two separate commands (`html-uri`, `latex-uri`) to make sure there can be different arguments and documentation.

Reply:
I don't think there is a way to tell cmdliner that a flag is allowed only when another flag is used, but I would be happy to be wrong :)

As you said in the other comment, there is also the problem of appending the right way: what happens if there is no trailing `/`, should we add one? Should we parse the prefix to look for error?

I see two possibilities:
- Either we rename `prefix` to `root_dir`, use cmdliner [dir](https://erratique.ch/software/cmdliner/doc/Cmdliner/Arg/index.html#val-dir) to check that such a directory exists, and do the "appending logic".
We also make it do nothing on latex uri (except warning the user that `root_dir` cannot be used in conjunction with latex). 
- Or, we get rid of it entirely.

Reply:
Sorry I somehow missed your comment...
> Perhaps it can be split into two separate commands (html-uri, latex-uri) to make sure there can be different arguments and documentation.
We can also check and fail if `html_root_dir` is present in Latex mode, ourselves, not relying on Cmdliner

Reply:
I don't prefer this solution because the error message will be different and that's new edge cases to document.

Reply:
In that case, I agree to have different commands

Reply:
I was not a fan of having several command, but I have come to agree that if it's the only way to have different set of options, that sounds the better option.

And I agree to add a `/` whenever there is no trailing slash to the prefix.